### PR TITLE
mapnik 3.0.12

### PR DESCRIPTION
**Pull request by springmeyer**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --new-formula <formula>` (after doing `brew install <formula>`)?

### CODE REVIEW

File: Formula/mapnik.rb [hunk 1]
```diff
@@ -84,4 +83,9 @@ def install
     system "make"
     system "make", "install"
   end
+
+  test do
+    assert_equal "#{prefix}".to_s, `#{bin}/mapnik-config --prefix`.chomp
```
Comment (zmwangx) [inline Formula/mapnik.rb]:
```ruby
assert_equal prefix.to_s, shell_output("#{bin}/mapnik-config --prefix").chomp
```

**Comment by zmwangx** [comment]
One comment, and please squash to one commit with the subject `mapnik 3.0.12` (please keep in mind that our preferred commit subject is `<formula name> <new version number>`). Otherwise 👍

**Comment by springmeyer** [comment]
@zmwangx thank you for the review. I've corrected the test syntax and squashed to a single commit. Is this acceptable now?

**Comment by zmwangx** [comment]
Thanks, I'm happy and let's just wait to make sure the CI is happy too.

File: Formula/mapnik.rb [hunk 2]
```diff
@@ -84,4 +83,9 @@ def install
     system "make"
     system "make", "install"
   end
+
+  test do
+    assert_equal prefix.to_s, shell_output("#{bin}/mapnik-config --prefix").chomp
+  end
+
```
Comment (ilovezfs) [inline Formula/mapnik.rb]: this line should not be here

File: Formula/mapnik.rb [hunk 3]
```diff
@@ -1,9 +1,8 @@
 class Mapnik < Formula
   desc "Toolkit for developing mapping applications"
   homepage "http://www.mapnik.org/"
-  url "https://github.com/mapnik/mapnik/archive/v3.0.9.tar.gz"
-  sha256 "f0242606096e2c4ca2cd0caac1ff0fd5f8054a38b5f288ba38b0e397b5b311b2"
-  revision 1
+  url "https://github.com/mapnik/mapnik/releases/download/v3.0.12/mapnik-v3.0.12.tar.bz2"
```
Comment (zmwangx) [inline Formula/mapnik.rb]: I was just about to pull this before I noticed another thing. You should not change the structure of this download URL, and in particular, it's conventional to use gzip tarballs over bz2 around here.

Comment (zmwangx) [inline Formula/mapnik.rb]: So https://github.com/mapnik/mapnik/archive/v3.0.12.tar.gz

Comment (ilovezfs) [inline Formula/mapnik.rb]: @zmwangx actually this is correct as-is. We like to use the release tarballs if they're available. Usually the problem is the reverse. We have to switch to the archive tarball temporarily for some reason, and then when that reason goes away, people forget to restore the release tarball until one of the maintainers notices there is a release tarball available and that it works again, and puts it back.

Comment (zmwangx) [inline Formula/mapnik.rb]: Oh I see what you're saying 👍

File: Formula/mapnik.rb [hunk 4]
```diff
@@ -1,9 +1,8 @@
...
+  sha256 "66a3d620c3ce543c91ea5b42a25079aca9a2a90f6079a2ce2a8714398fa57d6d"
```
Comment (ilovezfs) [inline Formula/mapnik.rb]: this blank line between the checksum and head should be removed. The blank line needs to be there if and only if there is a formula revision.

File: Formula/mapnik.rb (hunk 3 above)
Comment (ilovezfs) [inline Formula/mapnik.rb]: @zmwangx by the reason is primarily, though not solely, that the release tarballs will not require automake, autoconf, and libtool, whereas the source archive tarball may.

**Comment by zmwangx** [comment]
@BrewTestBot test this please

**Comment by ilovezfs** [comment]
@zmwangx No rest for the wicked.

**Comment by zmwangx** [comment]
@ilovezfs Converse is not true.

File: Formula/mapnik.rb (hunk 2 above)
Comment (zmwangx) [inline Formula/mapnik.rb]: Fixed this.

File: Formula/mapnik.rb (hunk 4 above)
Comment (zmwangx) [inline Formula/mapnik.rb]: Forgot to fix this... :trollface:

Comment (ilovezfs) [inline Formula/mapnik.rb]: tsk tsk.

**Comment by zmwangx** [comment]
Merged as 85253b3. Thank you for your contribution to Homebrew!

File: Formula/mapnik.rb (hunk 4 above)
Comment (ilovezfs) [inline Formula/mapnik.rb]: @zmwangx there I fixed it for you: https://github.com/Homebrew/homebrew-core/pull/4657

Comment (zmwangx) [inline Formula/mapnik.rb]: @ilovezfs  Kind of overkill but 👍 As a bonus, you might want to open another whitespace nazi PR to fix this :trollface: 

```
$ pcregrep -Ml '^  sha256.*\n\n^  head' *.rb | wc -l
238
```

Comment (ilovezfs) [inline Formula/mapnik.rb]: Be my guest.

File: Formula/mapnik.rb (hunk 3 above)
Comment (springmeyer) [inline Formula/mapnik.rb]: @ilovezfs - thanks that is exactly the situation here. I am the maintainer of Mapnik and made specific changes upstream to enable homebrew to get back on the release tarball. The archive version automatically created by github is not viable - it does not include all the things needed to built easily and we don't want people to use it.

As far as .gz vs .bz2 - I changed the .bz2 because that is all we provide upstream. I can provide .gz if that is important for some reason but I would prefer not to maintain two compression formats unless absolutely necessary.

So, to recap, the new url is necessary.

Comment (zmwangx) [inline Formula/mapnik.rb]: Thanks for the clarification. gzip is not necessary. It's just that when there are several archives with the same content, the gzip one is usually chosen (from my anecdotal observation).

Comment (springmeyer) [inline Formula/mapnik.rb]: 👍
